populator-machinery/controller.go

```diff
@@ -549,8 +624,42 @@ func (c *controller) syncPvc(ctx context.Context, key, pvcNamespace, pvcName str
 	}
 
 	// *** Here is the first place we start to create/modify objects ***
+	// If the PVC is unbound and PVC' doesn't exist yet, create PVC'
+	if "" == pvc.Spec.VolumeName && pvcPrime == nil {
+		pvcPrime = &corev1.PersistentVolumeClaim{
+			ObjectMeta: metav1.ObjectMeta{
+				Name:      pvcPrimeName,
+				Namespace: c.populatorNamespace,
+			},
+			Spec: corev1.PersistentVolumeClaimSpec{
+				AccessModes:      pvc.Spec.AccessModes,
+				Resources:        pvc.Spec.Resources,
+				StorageClassName: &storageClassPrimeName,
+				VolumeMode:       pvc.Spec.VolumeMode,
+			},
+		}
+		if waitForFirstConsumer {
+			pvcPrime.Annotations = map[string]string{
+				annSelectedNode: nodeName,
+			}
+		}
+		_, err = c.kubeClient.CoreV1().PersistentVolumeClaims(c.populatorNamespace).Create(ctx, pvcPrime, metav1.CreateOptions{})
+		if err != nil {
+			c.recorder.Eventf(pvc, corev1.EventTypeWarning, reasonPVCCreationError, "Failed to create populator PVC: %s", err)
+			return err
+		}
+		// We'll get called again later when the PVC' exists
```

Comment:
same here? At the end of a successful Create() call, we expect the resource to be ready. It is returned in the first arg of the return. We can move forward to the next step instead of returning null, I think

Reply:
I was following the style of https://github.com/kubernetes-csi/lib-volume-populator/blob/release-2.0/populator-machinery/controller.go#L609 If so seems like this is also not necessary?

Reply:
Right I don't see a reason why we need to return nil. If the object if created we get back the new object

Reply:
Ok, I'll remove return nil for PVC' and StorageClass' creation, and keep the existing Pod creation as it is